group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorShard.java

```diff
@@ -250,7 +251,7 @@ public GroupCoordinatorShard build() {
                 throw new IllegalArgumentException("TopicPartition must be set.");
             if (groupConfigManager == null)
                 throw new IllegalArgumentException("GroupConfigManager must be set.");
-            if (authorizer == null)
+            if (authorizerPlugin == null)
```

Comment:
Should we also check `authorizerPlugin.isEmpty()`?

Reply:
It's fine to pass `Optional.empty()` down the call stack.